test/unit/org/apache/cassandra/index/sai/cql/VectorSegmentationTest.java

```diff
@@ -0,0 +1,142 @@
+
+        int vectorCount = 100;
+        List<float[]> vectors = new ArrayList<>();
+        for (int row = 0; row < vectorCount; row++)
+        {
+            float[] vector = nextVector();
+            vectors.add(vector);
+            execute("INSERT INTO %s (pk, val) VALUES (?, ?)", row, vector(vector));
+        }
+
+        flush();
+
+        SegmentBuilder.updateLastValidSegmentRowId(17); // 17 rows per segment
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex'");
+
+        int limit = 35;
+        float[] queryVector = nextVector();
+        UntypedResultSet resultSet = execute("SELECT * FROM %s ORDER BY val ANN OF ? LIMIT " + limit, vector(queryVector));
+        assertThat(resultSet.size()).isEqualTo(limit);
+
+        List<float[]> resultVectors = getVectorsFromResult(resultSet);
+        double recall = rawIndexedRecall(vectors, queryVector, resultVectors, limit);
+        assertThat(recall).isGreaterThanOrEqualTo(0.99);
+    }
+
+    @Test
+    public void testMultipleSegmentsForCompaction() throws Throwable
+    {
+        createTable("CREATE TABLE %s (pk int, val vector<float, " + dimension + ">, PRIMARY KEY(pk))");
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex'");
+
+        List<float[]> vectors = new ArrayList<>();
+        int rowsPerSSTable = 10;
+        int sstables = 5;
+        int pk = 0;
+        for (int i = 0; i < sstables; i++)
+        {
+            for (int row = 0; row < rowsPerSSTable; row++)
+            {
+                float[] vector = nextVector();
+                execute("INSERT INTO %s (pk, val) VALUES (?, ?)", pk++, vector(vector));
+                vectors.add(vector);
+            }
+
+            flush();
+        }
+
+        int limit = 30;
+        float[] queryVector = nextVector();
+        UntypedResultSet resultSet = execute("SELECT * FROM %s ORDER BY val ANN OF ? LIMIT " + limit, vector(queryVector));
+        assertThat(resultSet.size()).isEqualTo(limit);
+
+        List<float[]> resultVectors = getVectorsFromResult(resultSet);
+        double recall = rawIndexedRecall(vectors, queryVector, resultVectors, limit);
+        assertThat(recall).isGreaterThanOrEqualTo(0.99);
+
+
+        SegmentBuilder.updateLastValidSegmentRowId(11); // 11 rows per segment
+        compact();
+
+        queryVector = nextVector();
+        resultSet = execute("SELECT * FROM %s ORDER BY val ANN OF ? LIMIT " + limit, vector(queryVector));
+        assertThat(resultSet.size()).isEqualTo(limit);
+
+        resultVectors = getVectorsFromResult(resultSet);
+        recall = rawIndexedRecall(vectors, queryVector, resultVectors, limit);
+        assertThat(recall).isGreaterThanOrEqualTo(0.99);
+    }
+
+    protected Vector<Float> vector(float[] values)
+    {
+        Float[] floats = new Float[values.length];
+        for (int i = 0; i < values.length; i++)
+            floats[i] = values[i];
+
+        return new Vector<>(floats);
+    }
```

Comment:
I think this method is overriding `CQLTester#vector(float[])` with a mostly identical implementation.

Reply:
I think the method is commented out, instead of just removed.